python/lsst/pipe/tasks/make_psf_matched_warp.py

```diff
@@ -0,0 +1,215 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import lsst.geom as geom
+import numpy as np
+import warnings
+
+from lsst.afw.geom import Polygon, makeWcsPairTransform
+from lsst.coadd.utils import copyGoodPixels
+from lsst.ip.diffim import ModelPsfMatchTask
+from lsst.meas.algorithms import GaussianPsfFactory, WarpedPsf
+from lsst.pex.config import ConfigurableField
+from lsst.pipe.base import (
+    PipelineTask,
+    PipelineTaskConfig,
+    PipelineTaskConnections,
+    Struct,
+)
+from lsst.pipe.base.connectionTypes import Input, Output
+from lsst.utils.timer import timeMethod
+
+if TYPE_CHECKING:
+    from lsst.afw.image import Exposure
+
+
+class MakePsfMatchedWarpConnections(
+    PipelineTaskConnections,
+    dimensions=("tract", "patch", "skymap", "instrument", "visit"),
+    defaultTemplates={
+        "coaddName": "deep",
+    },
+):
+    """Connections for MakePsfMatchedWarpTask"""
+
+    direct_warp = Input(
+        doc="Direct warped exposure produced by resampling calexps onto the skyMap patch geometry",
+        name="{coaddName}Coadd_directWarp",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "instrument", "visit"),
+    )
+
+    psf_matched_warp = Output(
+        doc=(
+            "Output PSF-Matched warped exposure, produced by resampling ",
+            "calexps onto the skyMap patch geometry and PSF-matching to a model PSF.",
+        ),
+        name="{coaddName}Coadd_psfMatchedWarp",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "visit", "instrument"),
+    )
+
+
+class MakePsfMatchedWarpConfig(
+    PipelineTaskConfig,
+    pipelineConnections=MakePsfMatchedWarpConnections,
+):
+    """Config for MakePsfMatchedWarpTask."""
+
+    modelPsf = GaussianPsfFactory.makeField(doc="Model Psf factory")
+    psfMatch = ConfigurableField(
+        target=ModelPsfMatchTask,
+        doc="Task to warp and PSF-match calexp",
+    )
+
+    # matchingKernelSize is there just for backward compatibility.
+    @property
+    def matchingKernelSize(self):
+        return self.psfMatch.kernel.active.kernelSize
+
+    @matchingKernelSize.setter
+    def matchingKernelSize(self, value):
+        self.psfMatch.kernel.active.kernelSize = value
```

Comment:
I think support for `matchingKernelSize` can be dropped altogether in this pipeline. We are supposed to directly set the `kernelSize` in the subtask instead of wrapping it as another config field. If we continue to support, at least we should warn that setting `matchingKernelSize` is not the recommended method.